mesa/time.py

```diff
@@ -327,3 +328,113 @@ def get_type_count(self, type_class: type[Agent]) -> int:
         Returns the current number of agents of certain type in the queue.
         """
         return len(self.agents_by_type[type_class])
+
+
+class DiscreteEventScheduler(BaseScheduler):
+    """
+    A scheduler for discrete event simulation in Mesa.
+
+    This scheduler manages events where each event is associated with a
+    specific time and agent. The scheduler advances time not in fixed
+    increments, but to the moment the next event is scheduled to occur.
+
+    This implementation uses a priority queue (heapq) to manage events. Each
+    event is a tuple of the form (time, random_value, agent), where:
+        - time (float): The scheduled time for the event.
+        - random_value (float): A secondary sorting criterion to randomize
```

Comment:
Would have been clearer to say that the `random_value` is used only once for the `heapq.push` operation. I had to check elsewhere in the code for its usage, until I realized of this.

Reply:
Read it for again, I think it’s quite clear, especially together with the comments lower in the code..